### DISCUSSION TITLE
minimal media question

### DISCUSSION THREAD

**Discussion by amarsh26** [opencobra/cobrapy#1317] [Q&A] [answered]
Hello,

When I manually enter the media from minimal_medium(model, max_growth), I receive a different biomass (0.8571429) then using tool to obtain minimal medium for FBA (0.876997). The max_growth used to obtain the minimal was the default. Is this expected behavior?

In addition, if I slightly increase the value for mobd in the manually entered minimal media from (0.000006  to 0.000007) and change no other values, the biomass increase to 0.8761609907. I was also wondering if this is expected. 

The data was obtained using the iML1515 model of _E. coli_.

Thanks!

**Comment by cdiener** [comment]
Sorry for the delayed response. By hand do you mean just typing the first x digits? Than this would be expected because this is not the exact value returned and you would be truncating it...

**Comment by amarsh26** [comment]
Hi,

Sorry for the late reply; I got pulled away from looking into this. Yes, I am taking the output produced from 
`model_default_growth = iml1515_model.slim_optimize() `
`minimal_medium(iml1515_model, model_default_growth)`
which is
 `EX_pi_e          0.845957
EX_mn2_e         0.000606
EX_fe2_e         0.014085
EX_glc__D_e     10.000000
EX_zn2_e         0.000299
EX_mg2_e         0.007608
EX_ca2_e         0.004565
EX_ni2_e         0.000283
EX_cu2_e         0.000622
EX_cobalt2_e     0.000022
EX_mobd_e        0.000006
EX_so4_e         0.220845
EX_nh4_e         9.471495
EX_k_e           0.171184
EX_cl_e          0.004565
EX_o2_e         22.131763`
and manually entering this data (including entering zero for compounds not in this output, but are occurring in the default media).

Thank you so much for your help!

**Comment by cdiener** [comment, marked as answer]
Yeah, if you manually enter the data that will introduce small errors what will change the output slightly. The default output will not show all digits of a float, so something like `0.10000023344` will become `0.1` there. If you just copy that you something that is somewhat but not exactly the same. If you assign directly without truncating it should be much closer, e. g.

```python
# ...
model_default_growth = iml1515_model.slim_optimize() 
med = minimal_medium(iml1515_model, model_default_growth) 
with iml1515_model as mod:
    mod.medium = med
    model_minimal_growth = mod.slim_optimize()
print(model_default_growth, model_minimal_growth)
```

Those values should be within the solver tolerance.

**Comment by amarsh26** [comment]
Thanks so much for your help!
